src/lib.rs

```diff
@@ -0,0 +1,286 @@
+
+use log::{info, error, LevelFilter};
+use log4rs::append::rolling_file::RollingFileAppender;
+use log4rs::encode::pattern::PatternEncoder;
+use log4rs::config::{Appender, Config, Root};
+use log4rs::filter::threshold::ThresholdFilter;
+use log4rs::append::rolling_file::policy::compound::{
+    CompoundPolicy,
+    trigger::size::SizeTrigger,
+    roll::fixed_window::FixedWindowRoller,
+};
+
+
+use zecwalletlitelib::{commands,
+    lightclient::{LightClient, LightClientConfig},
+};
+
+#[macro_export]
+macro_rules! configure_clapapp {
+    ( $freshapp: expr ) => {
+    $freshapp.version("1.0.0")
+            .arg(Arg::with_name("dangerous")
+                .long("dangerous")
+                .help("Disable server TLS certificate verification. Use this if you're running a local lightwalletd with a self-signed certificate. WARNING: This is dangerous, don't use it with a server that is not your own.")
+                .takes_value(false))
+            .arg(Arg::with_name("nosync")
+                .help("By default, zecwallet-cli will sync the wallet at startup. Pass --nosync to prevent the automatic sync at startup.")
+                .long("nosync")
+                .short("n")
+                .takes_value(false))
+            .arg(Arg::with_name("recover")
+                .long("recover")
+                .help("Attempt to recover the seed from the wallet")
+                .takes_value(false))
+            .arg(Arg::with_name("plantseed")
+                .short("s")
+                .long("plantseed")
+                .value_name("seed_phrase")
+                .help("Create a new wallet with the given 24-word seed phrase. Will fail if wallet already exists")
+                .takes_value(true))
+            .arg(Arg::with_name("server")
+                .long("server")
+                .value_name("server")
+                .help("Lightwalletd server to connect to.")
+                .takes_value(true)
+                .default_value(lightclient::DEFAULT_SERVER))
+            .arg(Arg::with_name("COMMAND")
+                .help("Command to execute. If a command is not specified, zecwallet-cli will start in interactive mode.")
+                .required(false)
+                .index(1))
+            .arg(Arg::with_name("PARAMS")
+                .help("Params to execute command with. Run the 'help' command to get usage help.")
+                .required(false)
+                .multiple(true))
+    };
+}
+
+pub fn report_permission_error() {
+    let user = std::env::var("USER").expect(
+        "Unexpected error reading value of $USER!");
+    let home = std::env::var("HOME").expect(
+        "Unexpected error reading value of $HOME!");
+    let current_executable = std::env::current_exe()
+        .expect("Unexpected error reporting executable path!");
+    eprintln!("USER: {}", user);
+    eprintln!("HOME: {}", home);
+    eprintln!("Executable: {}", current_executable.display());
+    if home == "/" {
```

Comment:
Does this work on Windows as well? The environment variables?

Reply:
Hmm...   I haven't tested.   Perhaps I should guarantee that this function is only called on POSIX?, unix? systems...   or look for a library that abstracts across OSs.